### DISCUSSION TITLE
tls: handshake failure on bootstrap

### DISCUSSION THREAD

**Discussion by jenshornan** [fluxcd/flux2#4998] [Q&A]
Hi!

I get the following error when I try to bootstrap flux. Running on k3s.

```
failed to checkout and determine revision: unable to clone 'https://github.com/xxxxxxx/starship.git': Get "https://github.com/xxxxxx/starship.git/info/refs?service=git-upload-pack": remote error: tls: handshake failure
```

This is the command:

```
flux bootstrap github \
  --token-auth \
  --owner=git \
  --repository=starship \
  --branch=main \
  --path=clusters/my-cluster \
  --personal
```

I have also set the GITHUB_TOKEN env variable.

Thx!

//Br Jens

**Comment by lejeunel** [comment]
I'm having the same issue using a gitlab enterprise repo.

 I have setup a GITLAB_TOKEN env variable which works fine since flux manages to push its manifests.
 The issue, similar to OP, arises when the flux-system/kustomize-controller attempts to clone the repository.
 
 For reference, my bootstrap command looks like the following:
 
 `
flux bootstrap gitlab --ca-file=<path-to-cert> --token-auth --hostname=<hostname-to-gitlab-server>  --owner=<group/subgroup> --repository=<my-repo> --branch=main --path=clusters/dev
`
